File: 496-next-greater-element-i/next-greater-element-i.js
var nextGreaterElement = function (nums1, nums2) {
    let res = [];
    for (let i = 0; i < nums1.length; i++) {
        let j = 0;
        while (nums1[i] != nums2[j]) j++;
        j++;
        while(j<nums2.length){
            if(nums2[j]>nums1[i]){
                res.push(nums2[j]);
                break;
            }
            j++;
        }
        if(!res[i]) res.push(-1);
    }
    return res;
};

File: 496-next-greater-element-i/next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        Map<Integer, Integer> m = new HashMap<Integer, Integer>();
        Stack<Integer> st = new Stack<Integer>();

        for(int num:nums2){
            while(!st.isEmpty() && st.peek()<num){
                m.put(st.pop(),num);
            }

            st.push(num);
        }

        int[] res = new int[nums1.length];
        int i = 0;
        for(int num:nums1){
            res[i++] = m.getOrDefault(num,-1);
        }

        return res;
    }
}